### PR TITLE
make istio-ca-root-cert in namespaces respect revision

### PR DESCRIPTION
**Pull request by irisdingbj**
currently, namespace controller will create the fixed `istio-ca-root-cert` in namespace even if you install istio with --revision.

This PR will support we have multi istio mesh in a single cluster. 

gateway need special logic. one way is  to let istiod create `istio-ca-root-cert` with revision in system namespace by default.  Will address this later.

### CODE REVIEW

**Comment by irisdingbj** [comment]
Thanks @shankgan as he pointed out this issue to me.

File: istioctl/cmd/workload.go [hunk 1]
```diff
@@ -333,7 +333,18 @@ func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, config *meshconfig.Proxy
 // the token is generated by kubectl under the workload group's namespace and service account
 // TODO: Make the following accurate when using the Kubernetes certificate signer
 func createCertsTokens(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, dir string, out io.Writer) error {
-	rootCert, err := kubeClient.CoreV1().ConfigMaps(wg.Namespace).Get(context.Background(), controller.CACertNamespaceConfigMap, metav1.GetOptions{})
+	ns, err := kubeClient.CoreV1().Namespaces().Get(context.Background(), wg.Namespace, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("namespace %s was not found in cluster %s", wg.Namespace, err)
```
Comment (shamsher31) [inline istioctl/cmd/workload.go:338]:
```suggestion
		return fmt.Errorf("namespace %s not found in the cluster %s", wg.Namespace, err)
```

File: istioctl/cmd/workload.go [hunk 2]
```diff
@@ -333,7 +333,18 @@ func createClusterEnv(wg *clientv1alpha3.WorkloadGroup, config *meshconfig.Proxy
...
+	}
+	var caCertNamespaceConfigMapWithRevision string
+	revision := ns.Labels["istio.io/rev"]
```
Comment (shamsher31) [inline istioctl/cmd/workload.go:341]: Is the `revision` label exists at the namespace level?

Comment (irisdingbj) [reply istioctl/cmd/workload.go:341]: it will exist only if you add it manually, so we will use the default cm in such case

**Review by howardjohn — CHANGES_REQUESTED** [review]
I have a few of concerns with this.

1. Revision is not namespace scoped, you can have N revisioned pods in one namespace. Writing the configmap in only some namespaces will break those mixed namespaces.
2. I think this will break upgrades
3. We have no design docs or discussion about multiple meshes

**Comment by shankgan** [comment]
Agree with @howardjohn. Also, could this PR could be tracked as part of a larger github issue to allow all IstiodCA related  components (The ca certs it depends on etc) to be revisioned? Given that this isn't a small change, I would recommend an  accompanying RFC to gather comments from the community.

**Comment by irisdingbj** [comment]
Thanks for the suggestions.  I will submit a design doc for this.  Let's me mark this PR `on-hold` for now until we sorted the design out.

**Comment by irisdingbj** [comment]
@howardjohn @shankgan @costinm  design doc is ready, PTAL : https://docs.google.com/document/d/1aLDfouW962hf8FP3poGOPOAlzBmiBYWrwoUhO9nD9hs/edit#  and looking forward to your comments!

**Pull request closed without merging**